aries_cloudagent/protocols/present_proof/v2_0/formats/dif/handler.py

```diff
@@ -0,0 +1,268 @@
+
+        """
+        dif_proof_request = V20PresProposal.deserialize(
+            pres_ex_record.pres_proposal
+        ).attachment(self.format)
+
+        return self.get_format_data(PRES_20_REQUEST, dif_proof_request)
+
+    async def create_pres(
+        self,
+        pres_ex_record: V20PresExRecord,
+        request_data: dict = {},
+    ) -> Tuple[V20PresFormat, AttachDecorator]:
+        """Create a presentation."""
+        proof_request = V20PresRequest.deserialize(
+            pres_ex_record.pres_request
+        ).attachment(self.format)
+        pres_definition = None
+
+        if request_data != {}:
+            pres_spec_payload = DIFPresSpecSpecSchema().load(request_data)
+            # Overriding with prover provided pres_spec
+            challenge = str(uuid4())
+            domain = None
+            pres_definition = pres_spec_payload.get("presentation_definition")
+            issuer_id = pres_spec_payload.get("issuer_id")
+        if not pres_definition:
+            challenge = proof_request.get("challenge") or str(uuid4())
+            domain = proof_request.get("domain") or None
+            pres_definition = PresentationDefinition.deserialize(
+                proof_request.get("presentation_definition")
+            )
+            issuer_id = None
+
+        input_descriptors = pres_definition.input_descriptors
+        try:
+            holder = self._profile.inject(VCHolder)
+            types = []
+            schema_ids = []
+            for input_descriptor in input_descriptors:
+                for schema in input_descriptor.schemas:
+                    uri = schema.uri
+                    required = schema.required or True
+                    if required:
+                        # JSONLD Expanded URLs
+                        if "#" in uri:
+                            types.append(uri.split("#")[1])
+                        else:
+                            schema_ids.append(uri)
+            if len(types) == 0:
+                types = None
+            if len(schema_ids) == 0:
+                schema_ids = None
+            search = holder.search_credentials(
+                types=types,
+                schema_ids=schema_ids,
+            )
+            # Defaults to page_size but would like to include all
+            # For now, setting to 1000
+            max_results = 1000
+            records = await search.fetch(max_results)
+        except StorageNotFoundError as err:
+            raise V20PresFormatError(err)
+        # Selecting suite from claim_format
+        claim_format = pres_definition.fmt
+        proof_type = None
+        if claim_format:
+            if claim_format.ldp_vp:
+                for proof_req in claim_format.ldp_vp.get("proof_type"):
+                    if proof_req == "BbsBlsSignature2020":
+                        proof_type = "BbsBlsSignature2020"
+                        break
+                    elif proof_req == "Ed25519Signature2018":
```

Comment:
Maybe change all string occurrences to the signature type property on the signature suite to avoid retyping the string value?

```suggestion
                    elif proof_req == Ed25519Signature2018.signature_type:
```